packages/dds/sequence/src/test/fuzz/sharedString.squash.fuzz.spec.ts

```diff
@@ -72,23 +72,19 @@ function makeExitingStagingModeGenerator() {
 		} = state;
 		assert(isPoisonedSharedString(sharedString));
 		const { poisonedHandleLocations } = sharedString;
-		// Mutating in the generator like this is not great practice, but it avoids having to define a special op type for 'removing poisoned content'
-		// which also handles removing values from `poisonedHandleLocations`.
-		// If debugging a situation where this test has strange behavior, consider adding that separate op type instead.
-		while (poisonedHandleLocations.length > 0) {
-			// safe due to length check above
-			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-			const lastPoisonedHandleLocation = poisonedHandleLocations.pop()!;
-			const pos = sharedString.localReferencePositionToPosition(lastPoisonedHandleLocation);
-			const segment = lastPoisonedHandleLocation.getSegment();
-			sharedString.removeLocalReferencePosition(lastPoisonedHandleLocation);
-			if (segment !== undefined && !segmentIsRemoved(segment)) {
-				return {
-					type: "removeRange",
-					start: pos,
-					end: pos + 1,
-				};
-			}
+		if (poisonedHandleLocations.length > 0) {
```

Comment:
the fuzz harness had a mixin which would unconditionally pull from the base generator even if the op wasn't going to be used. That didn't work with the old version of this logic due to the mutation (which as I previously wrote was a bad idea to do...). I fixed up the strategy we take here so that we're only mutating in reducers and also tweaked the harness so that it doesn't invoke generators it doesn't plan on using. Both approaches would have been valid fixes to the bug, but it felt right to make both changes.